changes/Hop_sequence (modular).py

```diff
@@ -0,0 +1,133 @@
+import math
+import pandas as pd
+
+# we create a class Point() that have attributes x any y and a function to find distance between those points.
+
+class Point():
+  def __init__(self, x, y):
+    self.x = x
+    self.y = y
+
+  def distance(self,b):
+    x_term = int(self.x) - int(b.x)
+    y_term = int(self.y) - int(b.y)
+    return math.sqrt((x_term * x_term) + (y_term * y_term))
+    
+
+# we craete a class hop_func() that have attributes as limit_range, max_stars, limiting_brightness, click
+# hence for every click this class have functions that return a hop which can be stored in a hop_sequence 
+# to generate a star plot.
+
+# this value is used to avoid returning of hops even when there is no nearby star (e.g. editor clicking
+# somewhere near the borders).
+
+# max_stars is the limit put to remove unnessecory faint stars from our star list
+
+# limiting_brightness is the minimum brightness below which we consider a star to be faint and is used to 
+# remove stars unnecessary faint stars if the no of stars go up by max_stars
+
+class hop_func():
+  
+  def __init__(self, limit_range, max_stars, limiting_brightness, x, y):
+    self.limit_range = limit_range
+    self.max_stars = max_stars
+    self.limiting_brightness = limiting_brightness
+    self.click = Point(x, y)
+
+  # the function list_star() returns a list of stars that lie 
+  # within the sqaure formed with the click as center
+  
+  def list_stars(self, data):
+    stars_catalogue = pd.DataFrame(columns = ['star','ra','dec','brightness'])
+
+    l_x_limit = int(self.click.x) - self.limit_range
+    r_x_limit = int(self.click.x) + self.limit_range
+    l_y_limit = int(self.click.y) - self.limit_range
+    r_y_limit = int(self.click.y) + self.limit_range
+  
+    for i in range(len(data)):    
```

Comment:
If the data is large and if the data is sorted, you need to use while statement rather than if statement. And as we're going to have larger data, it is advised to pass in data that is sorted. Having sorted data helps such that you don't need to search entire data, but just till a specific point.

Reply:
by which value should I sort by? ra/ dec/ brightness?